### ISSUE TITLE
Request: Show at most one notification per "watch cycle" + separator between "watch cycles"

### ISSUE THREAD

**Issue by lydell** [brunch/brunch#1604]
### Description

brunch's out-of-the-box system notifications for build errors in watch mode are awesome! However, they'd be even better if they were slightly less noisy.

This is all about making the brunch experience even better.

### Expected behavior

Any time the watcher triggers a recompilation (such as when saving, adding or removing a file), at most **one** notification should be shown.

### Actual behavior

One notification is shown for every error.

![screenshot from 2016-12-29 13-25-59](https://cloud.githubusercontent.com/assets/2142817/21543963/6486cc1c-cdcb-11e6-9aa9-c1fb9e1f6579.png)

- In the top middle of the screen shot (50% covered by the panel on the right), you can see what a notification looks like in my OS. The first notication is shown for a couple of seconds, then the second for a couple of seconds, then the third ... When I had to move a very commonly `require`d file in a real project, it took a _very_ long time for all the notifications to show up.

- In the panel on the right you can see how quickly the list of "unread" notifications fills up.

- (Regarding "Cannot read property 'split' of undefined", see #1614.)

Here's the test repo: https://github.com/lydell/brunch-notification-overflow

1. `npm install`
2. `brunch watch`
3. `mv app/utils.js app/helpers.js`

### Environment

1. Brunch: commit afefc5e7643f37918ed037e3862461624e483425
2. Node: 6.9.1
3. NPM: 3.10.8
4. Operating system: Ubuntu Gnome 16.04

### `package.json` contents

```json
{
  "private": true,
  "devDependencies": {
    "brunch": "github:brunch/brunch#afefc5e7643f37918ed037e3862461624e483425",
    "javascript-brunch": "^2.0.0"
  }
}
```

### brunch config contents

```javascript
module.exports = {
  files: {
    javascripts: {
      joinTo: 'app.js'
    }
  }
}
```

### What should be shown in the one notification?

Currently, error messages are shown in the notification. On my OS, only one line of text is shown in notifications, so for me the notifications only give a quick hint on what happened. I always switch to the terminal the first thing I do, to see the entire error messages.

Here are some thoughts:

- Possibly think about if there's a better message to show than (part) of the error messages.
- Possibly let the notification be exactly like today if there's only one error.
- Possibly notify only the first error exactly like today if there are more than one error (leaving the others to only show up in the terminal).
- Possibly show some summary in the notification, such as "3 files failed to compile".

### Related request: Separator between "watch cycles"

It would also be very helpful if a `─────────` line was emitted between each "watch cycle". Here's an example:

1. I start `brunch watch`
2. There are two errors.
3. I fix one of them, and save the file, which causes a new watch cycle.

Before:

```
$ brunch w
31 Dec 13:30:50 - error: Compiling of app/test.js failed.

   SyntaxError: app/test.js: super() is only allowed in a derived constructor
     2 |   constructor() {
     3 |     this.foo = 1
   > 4 |     super()
       |     ^
     5 |   }
     6 | }
     7 | 
 
31 Dec 13:30:50 - error: Compiling of app/test.scss failed.

   Error: Invalid CSS after "  background-color": expected "}", was ": white"
           on line 3 of app/test.scss
   >>   background-color: white
      --^
 
31 Dec 13:30:50 - info: compiled in 741 ms
31 Dec 13:31:02 - error: Compiling of app/test.scss failed.

   Error: Invalid CSS after "  background-color": expected "}", was ": white"
           on line 3 of app/test.scss
   >>   background-color: white
      --^
 
31 Dec 13:31:02 - info: compiled in 92 ms
```

After:

```
$ brunch w
31 Dec 13:30:50 - error: Compiling of app/test.js failed.

   SyntaxError: app/test.js: super() is only allowed in a derived constructor
     2 |   constructor() {
     3 |     this.foo = 1
   > 4 |     super()
       |     ^
     5 |   }
     6 | }
     7 | 
 
31 Dec 13:30:50 - error: Compiling of app/test.scss failed.

   Error: Invalid CSS after "  background-color": expected "}", was ": white"
           on line 3 of app/test.scss
   >>   background-color: white
      --^
 
31 Dec 13:30:50 - info: compiled in 741 ms
───────────────────────────────────────────────────────────────────────────────────────
31 Dec 13:31:02 - error: Compiling of app/test.scss failed.

   Error: Invalid CSS after "  background-color": expected "}", was ": white"
           on line 3 of app/test.scss
   >>   background-color: white
      --^
 
31 Dec 13:31:02 - info: compiled in 92 ms
```

I think that line really makes it easier to see what's left to fix, and what's history in the log.

Why is this related to this issue/request? Because it needs to be shown **once** per "watch cycle" (and not when starting the watcher).

**Comment by lydell** [comment]
(Note to email readers: I've made some edits to the issue description.)

**Comment by lydell** [comment]
I moved the two bugs in this issue out into separate issues, so that they don't get buried in this request :)

**Comment by paulmillr** [comment]
fixed in new loggy

**Comment by shvaikalesh** [comment]
It is not.

**Comment by shvaikalesh** [comment]
Should be fixed when #1664 is done.
